lib/puppet_x/msutter/templates/invoke_dsc_resource.erb

```diff
@@ -42,12 +42,13 @@ $invokeParams = @{
 <% dsc_parameters.each do |p| -%>
     <%= p.name.to_s.gsub(/^dsc_/,'') %> = <%= format_dsc_value(p.value) %>
 <% end -%>
-  }
-<% if resource[:dscmeta_module_name].casecmp('PSDesiredStateConfiguration') == 0 -%>
```

Comment:
The problem was that it *should* have been `!= 0`

It just so happened to work in my testing because I didn't validate against `x` style resources.

So I chose to rewrite this a little bit to not hardcode the `PSDesiredStateConfiguration` here, but to use a different heuristic.